google/cloud/storage/internal/rest/stub.cc

```diff
@@ -167,7 +197,7 @@ StatusOr<ListBucketsResponse> RestStub::ListBuckets(
       absl::StrCat("storage/", options.get<TargetApiVersionOption>(), "/b"));
   auto auth = AddAuthorizationHeader(options, builder);
   if (!auth.ok()) return auth;
-  request.AddOptionsToHttpRequest(builder);
+  AddOptionsToRequestBuilder(options, builder);
```

Comment:
I guess `request.AddOptionsToHttpRequest(builder)` is to add info from request to the builder, after change, is the info from request missed? [Example](https://github.com/googleapis/google-cloud-cpp/blob/f080acd0c88b15f1d505b391c60aa69ac29849ff/google/cloud/storage/internal/rest/request_builder_test.cc#L42-L58)